src/sentry/tasks/post_process.py

```diff
@@ -249,8 +256,10 @@ def post_process_group(
             event=event,
             primary_hash=kwargs.get("primary_hash"),
         )
-        with metrics.timer("tasks.post_process.delete_event_cache"):
-            event_processing_store.delete_by_key(cache_key)
+
+        if not has_workflow_owners:
+            with metrics.timer("tasks.post_process.delete_event_cache"):
```

Comment:
When `has_workflow_owners` is true, and the event has no groupid, this block should run. You could do `if not has_workflow_owners or not event.group_id`?

Reply:
Added the `or not event.group_id` here. This is actually how I had it originally but I thought it would have be better to isolate the "workflow owner" logic.